tests/detail/begin_statement_builder.cpp

```diff
@@ -0,0 +1,211 @@
+              "BEGIN ISOLATION LEVEL READ UNCOMMITTED"_s);
+
+
+    EXPECT_EQ(get_text(begin_statement_builder::build(make_options(transaction_options::mode = transaction_mode::read_write,
+                                                                   transaction_options::deferrability = deferrable))),
+              "BEGIN READ WRITE DEFERRABLE"_s);
+    EXPECT_EQ(get_text(begin_statement_builder::build(make_options(transaction_options::mode = transaction_mode::read_write,
+                                                                   transaction_options::deferrability = !deferrable))),
+              "BEGIN READ WRITE NOT DEFERRABLE"_s);
+    EXPECT_EQ(get_text(begin_statement_builder::build(make_options(transaction_options::mode = transaction_mode::read_write))),
+              "BEGIN READ WRITE"_s);
+
+    EXPECT_EQ(get_text(begin_statement_builder::build(make_options(transaction_options::mode = transaction_mode::read_only,
+                                                                   transaction_options::deferrability = deferrable))),
+              "BEGIN READ ONLY DEFERRABLE"_s);
+    EXPECT_EQ(get_text(begin_statement_builder::build(make_options(transaction_options::mode = transaction_mode::read_only,
+                                                                   transaction_options::deferrability = !deferrable))),
+              "BEGIN READ ONLY NOT DEFERRABLE"_s);
+    EXPECT_EQ(get_text(begin_statement_builder::build(make_options(transaction_options::mode = transaction_mode::read_only))),
+              "BEGIN READ ONLY"_s);
+
+    EXPECT_EQ(get_text(begin_statement_builder::build(make_options(transaction_options::deferrability = deferrable))),
+              "BEGIN DEFERRABLE"_s);
+    EXPECT_EQ(get_text(begin_statement_builder::build(make_options(transaction_options::deferrability = !deferrable))),
+              "BEGIN NOT DEFERRABLE"_s);
+    EXPECT_EQ(get_text(begin_statement_builder::build(make_options())),
+              "BEGIN"_s);
+}
+
+TEST(begin_statement_builder, should_treat_none_like_non_existent_parameters) {
+    using namespace ozo;
+    using namespace ozo::detail;
+
+    EXPECT_EQ(get_text(begin_statement_builder::build(make_options(transaction_options::isolation_level = ozo::none))),
+              get_text(begin_statement_builder::build(make_options())));
+
+    EXPECT_EQ(get_text(begin_statement_builder::build(make_options(transaction_options::deferrability = ozo::none,
+                                                                   transaction_options::mode = transaction_mode::read_only))),
+              get_text(begin_statement_builder::build(make_options(transaction_options::mode = transaction_mode::read_only))));
+}
+
+struct my_true_t { constexpr static bool value = true; } my_true;
+
+TEST(begin_statement_builder, should_allow_integral_constants_for_deferrability) {
```

Comment:
I'm not sure about this test is necessary. You provide `deferrable` expectations in the previous test. I do not think that check for support custom true or false type is really needed.